WordPress/Classes/ViewRelated/Likes/LikesListController.swift

```diff
@@ -0,0 +1,343 @@
+            delegate?.didSelectHeader()
+            return
+        }
+
+        guard !isLoadingContent,
+              !isShowingError,
+              indexPath.row < likingUsers.count else {
+            return
+        }
+
+        delegate?.didSelectUser(likingUsers[indexPath.row])
+    }
+
+}
+
+// MARK: - Notification Cell Handling
+
+private extension LikesListController {
+
+    func headerCell() -> NoteBlockHeaderTableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: NoteBlockHeaderTableViewCell.reuseIdentifier()) as? NoteBlockHeaderTableViewCell,
+              let group = notification?.headerAndBodyContentGroups[Constants.headerRowIndex] else {
+            DDLogError("Error: couldn't get a header cell or FormattableContentGroup.")
+            return NoteBlockHeaderTableViewCell()
+        }
+
+        setupHeaderCell(cell: cell, group: group)
+        return cell
+    }
+
+    func setupHeaderCell(cell: NoteBlockHeaderTableViewCell, group: FormattableContentGroup) {
+        cell.attributedHeaderTitle = nil
+        cell.attributedHeaderDetails = nil
+
+        guard let gravatarBlock: NotificationTextContent = group.blockOfKind(.image),
+            let snippetBlock: NotificationTextContent = group.blockOfKind(.text) else {
+                return
+        }
+
+        cell.attributedHeaderTitle = formatter.render(content: gravatarBlock, with: HeaderContentStyles())
+        cell.attributedHeaderDetails = formatter.render(content: snippetBlock, with: HeaderDetailsContentStyles())
+
+        // Download the Gravatar
+        let mediaURL = gravatarBlock.media.first?.mediaURL
+        cell.downloadAuthorAvatar(with: mediaURL)
+    }
+
+    func userCell(for indexPath: IndexPath) -> NoteBlockUserTableViewCell {
+        guard indexPath.row < likingUsers.count,
+              let cell = tableView.dequeueReusableCell(withIdentifier: NoteBlockUserTableViewCell.reuseIdentifier()) as? NoteBlockUserTableViewCell else {
+            DDLogError("Error: couldn't get a user cell or requested row is out of boundary")
+            return NoteBlockUserTableViewCell()
+        }
+
+        let user = likingUsers[indexPath.row]
+        cell.accessoryType = .none
+        cell.name = user.displayName
+
+        // TODO: Re-enable follow functionality once the information is available.
+        cell.isFollowEnabled = false
+        cell.isFollowOn = false
+
+        // TODO: Configure blog title once the information is available.
+        cell.blogTitle = ""
+
+        if let mediaURL = URL(string: user.avatarURL) {
+            cell.downloadGravatarWithURL(mediaURL)
+        }
+
+        // configure the ending separator line
+        cell.isLastRow = (indexPath.row == likingUsers.count - 1)
+
+        return cell
+    }
+
+}
+
+// MARK: - Dependency Definitions
+
+/// A simple, convenient dependency wrapper for LikesListController.
+class LikesListDependency {
```

Comment:
Thanks for moving this, but I think my question wasn't clear. I'm confused as to why we have this class period. Its content is simply properties used by `LikesListController`. Why not just make them properties of `LikesListController`?

Reply:
Got it. Initially I thought it would be helpful for unit tests, but I guess I can remove this class instead. Thanks for the input!

Reply:
Ah yes, initially I thought it would be helpful for unit testing later on, but I guess I can remove it to simplify the class. Thanks for the input!